### PR TITLE
Add an example for MultiNodeWeightedSampler 

### LINKED ISSUE #1352: [WIP] Examples for demonstrating the usage and incremental value of TorchData Nodes
### 🚀 The feature

Starting this issue to track minimal examples we can create to demonstrate effective usage and value of TorchData nodes. I can create separate issues for each of these as required.

### Motivation, pitch

1. Vanilla torch.utils dataloader usage ported over to torchdata nodes
2. GPU accelerated transforms
3. Flexible parallelism (mixing multiprocessing with multithreading)
4. Examples porting over popular OSS datasets
    -  connecting to popular cloud storage
5. Example creating new nodes (might get covered through examples above)
6. Basic multimodal model trained E2E using torchdata nodes
7. Chaining multiple transforms (might get covered through examples above)
8. Dataset mixing (with different sampling strategies)

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ramanishsingh**
Added a very simple example for showing the functionality for MultiNodeWeightedSampler
Fixes #1352

### CODE REVIEW

**Comment by divyanshk** [comment]
Super Duper Nit: can you add a space between the comments' hashtag and the first letter (makes things consistent and neater), thank you :-)

**Comment by divyanshk** [comment]
For the first cell, we can make the contents show up in a markdown or text cell? instead of comments in a code cell

**Comment by divyanshk** [comment]
Also, the three core lines of the notebook (sampler -> batcher -> loader) can be made to be in one cell? wdyt

**Review by divyanshk — APPROVED** [review]
Overall LGTM! Added some minor formatting requests.

**Comment by divyanshk** [comment]
Is this ready to land ? @ramanishsingh @andrewkho

**Comment by andrewkho** [comment]
Not yet, I think it can be simplified

**Comment by ramanishsingh** [comment]
> Not yet, I think it can be simplified
@andrewkho which parts do you think are still convoluted?

**Comment by andrewkho** [comment]
Whoops sorry I thought this was the RoundRobin sampler diff, let me have another look

File: examples/nodes/multi_dataset_weighted_sampling.ipynb [hunk 1]
```diff
@@ -0,0 +1,134 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "79a14c63-a085-493f-8db9-6af3e1d744b5",
+   "metadata": {},
+   "source": [
+    "### In this notebook, we will explore the usage of MultiNodeWeightedSampler in torchdata.nodes\n",
```
Comment (andrewkho) [inline examples/nodes/multi_dataset_weighted_sampling.ipynb:8]: ditto here 
```suggestion
    "In this notebook, we will explore the usage of MultiNodeWeightedSampler in torchdata.nodes\n",
```

Comment (andrewkho) [inline examples/nodes/multi_dataset_weighted_sampling.ipynb:8]: Add a title though, eg 
```suggestion
    "#MultiNodeWeightedSampler example\n",
    "In this notebook, we will explore the usage of MultiNodeWeightedSampler in torchdata.nodes\n",
```

File: examples/nodes/multi_dataset_weighted_sampling.ipynb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    "\n",
+    "# Next, we have to define weights for sampling from a particular dataset\n",
+    "weights = {\"ds0\": 0.5, \"ds1\": 0.25, \"ds2\": 0.25}\n",
+    "\n",
+    "# Finally we instatiate the MultiNodeWeightedSampler to sample from our datasets\n",
+    "multi_node_sampler = MultiNodeWeightedSampler(datasets, weights)\n",
+    "\n",
+    "# Since nodes are iterators, they need to be manually .reset() between epochs.\n",
+    "# We can wrap the root node in Loader to convert it to a more conventional Iterable.\n",
+    "train_loader = Loader(multi_node_sampler)"
```
Comment (andrewkho) [inline examples/nodes/multi_dataset_weighted_sampling.ipynb:49]: drop the train

File: examples/nodes/multi_dataset_weighted_sampling.ipynb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    "print(f\"The original weights were = {weights}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d1eb99df-7c62-47a4-b6c0-233a85f2e453",
+   "metadata": {},
+   "source": [
+    "#### Since picking items from the datasets according to weights is a stochastic process, the fraction of datasets in the items we have fetched is close to provided weights, but not exactly equal.\n",
+    "#### If we fetch more items, the fractions will asymptotically reach the provided weights."
```
Comment (andrewkho) [inline examples/nodes/multi_dataset_weighted_sampling.ipynb:84]: These don't need to be subheadings either

Comment (andrewkho) [reply examples/nodes/multi_dataset_weighted_sampling.ipynb:84]: actually I don't think we need this section at all, it seems a bit redundant. You can just use 100k in the cell above

**Review by andrewkho — CHANGES_REQUESTED** [review]
Few minor comments, but otherwise looking good!

**Review by andrewkho — APPROVED** [review]
gogogo

**Pull request merged**